Assets/MixedRealityToolkit/_Core/Definitions/Utilities/InscribedRectangle.cs

```diff
@@ -0,0 +1,570 @@
+            {
+                Edge edge = geometryEdges[i];
+
+                if ((edge.PointA.x < minX) || (edge.PointB.x < minX))
+                {
+                    minX = Mathf.Min(edge.PointA.x, edge.PointB.x);
+                }
+
+                if ((edge.PointA.y < minY) || (edge.PointB.y < minY))
+                {
+                    minY = Mathf.Min(edge.PointA.y, edge.PointB.y);
+                }
+
+                if ((edge.PointA.x > maxX) || (edge.PointB.x > maxX))
+                {
+                    maxX = Mathf.Max(edge.PointA.x, edge.PointB.x);
+                }
+
+                if ((edge.PointA.y > maxY) || (edge.PointB.y > maxY))
+                {
+                    maxY = Mathf.Min(edge.PointA.y, edge.PointB.y);
+                }
+
+            }
+
+            // Generate random points until we have randomPointCount starting points
+            Vector2[] startingPoints = new Vector2[randomPointCount];
+            {
+                System.Random random = new System.Random(randomSeed);
+                for (int i = 0; i < startingPoints.Length; i++)
+                {
+                    Vector2 candidatePoint;
+
+                    do
+                    {
+                        candidatePoint.x = ((float)random.NextDouble() * (maxX - minX)) + minX;
+                        candidatePoint.y = ((float)random.NextDouble() * (maxY - minY)) + minY;
+                    }
+                    while (!EdgeUtilities.IsInsideBoundary(geometryEdges, candidatePoint));
+
+                    startingPoints[i] = candidatePoint;
+                }
+
+            }
+
+            // Angles to use for fitting the rectangle within the boundary.
+            float[] angles = { 0, 15, 30, 45, 60, 75, 90, 105, 120, 135, 150, 165 };
+
+            for (int angleIndex = 0; angleIndex < angles.Length; angleIndex++)
+            {
+                for (int pointIndex = 0; pointIndex < startingPoints.Length; pointIndex++)
+                {
+                    Vector2 topCollisionPoint;
+                    Vector2 bottomCollisionPoint;
+                    Vector2 leftCollisionPoint;
+                    Vector2 rightCollisionPoint;
+
+                    float angleRadians = MathUtils.DegreesToRadians(angles[angleIndex]);
+
+                    // Find the collision point of a cross through the given point at the given angle.
+                    // Note, we are ignoring the return value as we are checking each point's validity
+                    // individually.
+                    FindSurroundingCollisionPoints(
+                        geometryEdges, 
+                        startingPoints[pointIndex], 
+                        angleRadians,
+                        out topCollisionPoint, 
+                        out bottomCollisionPoint, 
+                        out leftCollisionPoint, 
+                        out rightCollisionPoint);
+
+                    float newWidth = 0;
```

Comment:
why set to 0 if the values are just going to get overwritten?

Reply:
was in the original code. i left as much unchanged as possible